libminifi/src/core/json/JsonFlowSerializer.cpp

```diff
@@ -47,14 +47,21 @@ void JsonFlowSerializer::encryptSensitiveProperties(rapidjson::Value &property_j
     }
     if (properties.at(name).isSensitive()) {
       auto& value = property.value;
-      const std::string_view value_sv{value.GetString(), value.GetStringLength()};
+      const std::string_view value_sv = component_overrides.contains(name) ? component_overrides.at(name) : std::string_view{value.GetString(), value.GetStringLength()};
       const std::string encrypted_value = utils::crypto::property_encryption::encrypt(value_sv, encryption_provider);
       value.SetString(encrypted_value.c_str(), encrypted_value.size(), alloc);
     }
+    component_overrides.erase(name);
+  }
+
+  for (const auto& [name, value] : component_overrides) {
+    const std::string encrypted_value = utils::crypto::property_encryption::encrypt(value, encryption_provider);
+    property_jsons.AddMember(rapidjson::Value(name, alloc), rapidjson::Value(encrypted_value, alloc), alloc);
```

Comment:
What's the idea behind overrides? I see that the override values are used in plaintext to override existing properties in the flow definition, but if they were not used, then they are used encrypted. I don't understand why it works that way.

Reply:
Encrypted or not is a red herring. The property value in the JSON (or YAML) may or may not be already encrypted; if it isn't, then it will be encrypted now.  Overrides are used by the `encrypt-config` tool to replace the existing property value with a new value given by the user.  "Override" may not be the best name for this; do you have a suggestion for a better name?

Reply:
Also, this part of the code handles the case when the property given by the user to the `encrypt-config` tool does not exist in the JSON, yet.  In this case, we add the property to the JSON flow configuration.

Reply:
Overrides as a name is fine for me, but their expected format needs to be clearly defined.
I think we should standardize whether the values in the overrides are already encrypted (if sensitive) or not encrypted. I'm not sure how we currently decide about encryption based on this control flow.

Reply:
My bad, I missed the encrypt call on line 51 at first.